### ISSUE TITLE
[a11y]: ai-chat should announce dynamic content updates

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#877]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

Screen reader is not announcing the following content instantly from the Chat window, leaving screen reader users with no clue about these updates:
1. Screen reader does not announce status updates in the Chat window instantly, for example: “Thinking,” “Loading,” etc.
2. Screen reader does not announce new AI responses (answers) instantly when they appear.

Recommended fix: All the dynamic content updates must be instantly  conveyed to screen reader users , in accordance to its importance:
1. Implement live region Role and  properties such as aria-live="polite" or aria-live="assertive" accordingly , to ensure status changes (e.g., “Thinking,” “Loading”) and AI responses
are announced promptly by screen readers.

**Note 1:** Recommend to give more   clear user hints : for example -  Your AI Assistance is Thinking, Your AI Assistance responded "  etc.  
**Note 2:**  Ensure that the live region is scoped properly to only announce relevant updates (avoid repeating static or non-essential text).
**Note 3:** Confirm  the behavior  with multiple screen readers (JAWS, NVDA, Voiceover, Talkback) to.

User impact: Screen reader users are impacted because dynamic content changes,   AI responses , status updates, are not announced, making difficult to use the  Chat Assistance itself. 

### WCAG 2.1 Violation

WCAG Mapping: 4.1.2 Name, Role, Value

### Reproduction/example

n/a

### Steps to reproduce

1. open ai-chat
2. ask a question
3. see that the dynamic change of contents to "Thinking" is not announced
4. see that the new answer is not announced

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
